src/components/Modal/ReanimatedModal/utils.ts

```diff
@@ -0,0 +1,67 @@
+import {Easing} from 'react-native-reanimated';
+import type {ValidKeyframeProps} from 'react-native-reanimated/lib/typescript/commonTypes';
+import type {AnimationInType, AnimationOutType} from './types';
+
+const easing = Easing.bezier(0.76, 0.0, 0.24, 1.0).factory();
+
+function getModalInAnimation(animationType: AnimationInType): ValidKeyframeProps {
+    switch (animationType) {
+        case 'slideInRight':
+            return {
+                from: {transform: [{translateX: '100%'}]},
+                to: {
+                    transform: [{translateX: '0%'}],
+                    easing,
+                },
+            };
+        case 'slideInUp':
+            return {
+                from: {transform: [{translateY: '100%'}]},
+                to: {
+                    transform: [{translateY: '0%'}],
+                    easing,
+                },
+            };
+        case 'fadeIn':
+        default:
+            return {
+                from: {opacity: 0},
+                to: {
+                    opacity: 0.72,
+                    easing,
+                },
+            };
```

Comment:
NAB: Split 'fadeIn' and default to make the logic clearer and better catch unintended animation types

```suggestion
        case 'fadeIn':
         return {
                from: {opacity: 0},
                to: {
                    opacity: 0.72,
                    easing,
                },
            };
        
        default:
          throw ...
           
```